### PR TITLE
DFR-3460-Refactor-New-PaperCase

### PR DESCRIPTION
**Pull request by dawudgovuk**
### Jira link
https://tools.hmcts.net/jira/browse/DFR-3460

### Change description

Refactor New Paper Case about-to-submit handler

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/PaperCaseCreateContestedAboutToSubmitHandlerTest.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        GenericAboutToStartOrSubmitCallbackResponse<FinremCaseData> handle = handler.handle(callbackRequest, AUTH_TOKEN);
+        FinremCaseData caseData = handle.getData();
+
+        assertEquals(caseData.getContactDetailsWrapper().getIsAdmin(), NO_VALUE);
+        assertEquals(caseData.getFastTrackDecision(), YesOrNo.NO);
+        assertEquals(caseData.getPaperApplication(), YesOrNo.YES);
+        assertTrue(caseData.isApplicantRepresentedByASolicitor());
+    }
+
+}
```
Comment (ptrelease) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/PaperCaseCreateContestedAboutToSubmitHandlerTest.java:128]: New line

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/PaperCaseCreateContestedAboutToSubmitHandler.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        this.idamService = idamService;
+    }
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, CaseType caseType, EventType eventType) {
+        return CallbackType.ABOUT_TO_START.equals(callbackType)
+            && CaseType.CONTESTED.equals(caseType)
+            && (EventType.NEW_PAPER_CASE.equals(eventType));
+    }
+
```
Comment (ptrelease) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/PaperCaseCreateContestedAboutToSubmitHandler.java:43]: This won't run. CallbackType is wrong.

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/PaperCaseCreateContestedAboutToSubmitHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    void canNotHandle() {
+        assertFalse(handler
+                .canHandle(CallbackType.ABOUT_TO_START, CaseType.CONSENTED, EventType.NEW_PAPER_CASE));
+    }
+
+    @Test
+    void canNotHandleWrongEventType() {
+        assertFalse(handler
+                .canHandle(CallbackType.ABOUT_TO_START, CaseType.CONTESTED, EventType.CLOSE));
+    }
```
Comment (ptrelease) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/PaperCaseCreateContestedAboutToSubmitHandlerTest.java:63]: Ah, this is why your tests passed.  Change the CallbackType.

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/PaperCaseCreateContestedAboutToSubmitHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    @Mock
+    private CaseFlagsService caseFlagsService;
+
+    @BeforeEach
+    void setup() {
+        FinremCaseDetailsMapper finremCaseDetailsMapper = new FinremCaseDetailsMapper(new ObjectMapper().registerModule(new JavaTimeModule()));
+        handler = new PaperCaseCreateContestedAboutToSubmitHandler(finremCaseDetailsMapper, caseFlagsService, idamService);
+    }
+
+    @Test
```
Comment (jthmcts) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/PaperCaseCreateContestedAboutToSubmitHandlerTest.java:47]: When you're writing tests for the canHandle method you should use

https://github.com/hmcts/finrem-case-orchestration-service/blob/master/src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/test/Assertions.java

See here for example usage:
https://github.com/hmcts/finrem-case-orchestration-service/blob/1428fdb8d0471c84af48a926f0ba6b3814f2ac00/src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/draftorders/upload/UploadDraftOrderAboutToStartHandlerTest.java#L59

**Review by jthmcts — COMMENTED** [review]
What about the existing code controller? Ideally remove it or at least mark it as deprecated

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/PaperCaseCreateContestedAboutToSubmitHandlerTest.java [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.OrchestrationConstants.NO_VALUE;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.OrchestrationConstants.YES_VALUE;
+
+@ExtendWith(MockitoExtension.class)
+class PaperCaseCreateContestedAboutToSubmitHandlerTest extends BaseHandlerTestSetup {
+
+    private static final String CONTESTED_HWF_JSON = "/fixtures/contested/hwf.json";
+    private static final String CONTESTED_VALIDATE_HEARING_SUCCESSFULLY_JSON = "/fixtures/contested/validate-hearing-successfully.json";
+    private static final String AUTH_TOKEN = "tokien:)";
```
Comment (so99y) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/PaperCaseCreateContestedAboutToSubmitHandlerTest.java:34]: Use TestConstants.AUTHO_TOKEN instead.

**Comment by ptrelease** [comment]
> What about the existing controller code and tests? Ideally remove it or at least mark it as deprecated

@dawudgovuk Does this still need to be done?  I still see the controller method on the branch.

https://github.com/hmcts/finrem-case-orchestration-service/blob/2bac1a89749f49e0debdfff9dac1c9631e335eb0/src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/CaseDataController.java#L126

**Review by jthmcts — APPROVED** [review]

**Review by ptrelease — APPROVED** [review]

**Pull request merged**